src/metadata.cpp

```diff
@@ -796,10 +796,12 @@ Metadata::SchemaSnapshot Metadata::schema_snapshot(int protocol_version, const V
 }
 
 void Metadata::update_keyspaces(int protocol_version, const VersionNumber& cassandra_version, ResultResponse* result) {
+  // update_keyspaces may race with update_partitions.
+  // Obtain lock via mutex_ to protect schema_snapshot_version_.
+  ScopedMutex l(&mutex_);
   schema_snapshot_version_++;
 
   if (is_front_buffer()) {
-    ScopedMutex l(&mutex_);
```

Comment:
Disagree with the change.
Currently `mutex_` is protecting only the `front_` instance of `InternalData`.
If you would change it - you need to do it in ALL `update*` methods.

Alternative way - is to add another mutex for `back_` object. Not sure if we really need this.

We can try to protect only the `updating_` pointer value to prevent the error as I hope.

Reply:
This looks non standard:
```
void Metadata::update_partitions(int protocol_version, const VersionNumber& cassandra_version, ResultResponse* result) {
  ScopedMutex l(&mutex_);
  schema_snapshot_version_++;
  updating_->update_partitions(protocol_version, cassandra_version, cache_, result);
}
```
Compare with a standard implementation:
```
void Metadata::update_keyspaces(int protocol_version, const VersionNumber& cassandra_version, ResultResponse* result) {
  schema_snapshot_version_++;

  if (is_front_buffer()) {
    ScopedMutex l(&mutex_);
    updating_->update_keyspaces(protocol_version, cassandra_version, result);
  } else {
    updating_->update_keyspaces(protocol_version, cassandra_version, result);
  }
}
```